### PR TITLE
update ext-authz API comment

### PR DESCRIPTION
**Pull request by yangminzhu**
Update the comment to match the latest implementation.

cc @howardjohn @louiscryan

### CODE REVIEW

File: mesh/v1alpha1/config.proto [hunk 1]
```diff
@@ -520,9 +520,12 @@ message MeshConfig {
     // $hide_from_docs
     message EnvoyExternalAuthorizationHttpProvider {
       // REQUIRED. Specifies the service that implements the Envoy ext_authz HTTP authorization service.
-      // The format is "[<Namespace>/]<Service>". If the <Namespace> is omitted then it is resolved within the same
-      // namespace as this configuration resource. The <Service> is the name of the service object (k8s service or ServiceEntry).
-      // Example: "foo/my-ext-authz" or "my-ext-authz".
+      // The format is "[<Namespace>/]<Hostname>". The <Hostname> is the full qualified host name in the Istio service
+      // registry defined by the k8s service or ServiceEntry. The <Namespace> is the namespace of the k8s service or
```
Comment (howardjohn) [inline mesh/v1alpha1/config.proto:524]: nit: prefer "Kubernetes" to "k8s"

**Review by howardjohn — APPROVED** [review]

File: mesh/v1alpha1/config.proto [hunk 2]
```diff
@@ -567,9 +570,12 @@ message MeshConfig {
     // $hide_from_docs
     message EnvoyExternalAuthorizationGrpcProvider {
       // REQUIRED. Specifies the service that implements the Envoy ext_authz gRPC authorization service.
-      // The format is "[<Namespace>/]<Service>". If the <Namespace> is omitted then it is resolved within the same
-      // namespace as this configuration resource. The <Service> is the name of the service object (k8s service or ServiceEntry).
-      // Example: "foo/my-ext-authz" or "my-ext-authz".
+      // The format is "[<Namespace>/]<Hostname>". The <Hostname> is the full qualified host name in the Istio service
```
Comment (incfly) [inline mesh/v1alpha1/config.proto:573]: Allowing namespace specified, won't this contradict to the claim that we want only mesh admin be able to configure ext authz endpoint?

(Background reason for this, IIUC, is to accommodate the restrictions where mesh config is defined via file, not a config map in a certain namespace -- i don't know how that's possible though, btw)

File: mesh/v1alpha1/config.proto [hunk 3]
```diff
@@ -520,9 +520,12 @@ message MeshConfig {
     // $hide_from_docs
     message EnvoyExternalAuthorizationHttpProvider {
       // REQUIRED. Specifies the service that implements the Envoy ext_authz HTTP authorization service.
-      // The format is "[<Namespace>/]<Service>". If the <Namespace> is omitted then it is resolved within the same
-      // namespace as this configuration resource. The <Service> is the name of the service object (k8s service or ServiceEntry).
-      // Example: "foo/my-ext-authz" or "my-ext-authz".
+      // The format is "[<Namespace>/]<Hostname>". The <Hostname> is the full qualified host name in the Istio service
+      // registry defined by the k8s service or ServiceEntry. The <Namespace> is the namespace of the k8s service or
+      // ServiceEntry object, and can be omitted if the <Hostname> alone can decide the service unambiguously (normally
```
Comment (louiscryan) [inline mesh/v1alpha1/config.proto]: s/decide/resolve

File: mesh/v1alpha1/config.proto [hunk 4]
```diff
@@ -520,9 +520,12 @@ message MeshConfig {
     // $hide_from_docs
     message EnvoyExternalAuthorizationHttpProvider {
       // REQUIRED. Specifies the service that implements the Envoy ext_authz HTTP authorization service.
-      // The format is "[<Namespace>/]<Service>". If the <Namespace> is omitted then it is resolved within the same
-      // namespace as this configuration resource. The <Service> is the name of the service object (k8s service or ServiceEntry).
-      // Example: "foo/my-ext-authz" or "my-ext-authz".
+      // The format is "[<Namespace>/]<Hostname>". The <Hostname> is the full qualified host name in the Istio service
```
Comment (louiscryan) [inline mesh/v1alpha1/config.proto:523]: I would reword a little bit because you say 'full qualified' but then talk about qualifying it further....

The format is "[<Namespace>/]<Hostname>".  Specification of <Namespace> is required only when <Hostname> is insufficient to unambiguously resolve a service in the service registry.

**Review by louiscryan — APPROVED** [review]
LGTM with comments

**Comment by yangminzhu** [comment]
oh, the PR is merged a bit too quick, will send out another PR to address comments from Louis.
